spirv_msl.cpp

```diff
@@ -9626,6 +9656,75 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		break;
 	}
 
+	case OpSDot:
+	case OpUDot:
+	case OpSUDot:
+	{
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		uint32_t vec1 = ops[2];
+		uint32_t vec2 = ops[3];
+
+		auto& input_type1 = expression_type(vec1);
+		auto& input_type2 = expression_type(vec2);
+
+		string vec1input, vec2input;
+		if ((input_type1.basetype == SPIRType::BaseType::UInt || input_type1.basetype == SPIRType::BaseType::Int) && input_type1.vecsize == 1)
+		{
+			string type = opcode == OpSDot || opcode == OpSUDot ? "char4" : "uchar4";
+			vec1input = join("as_type<", type, ">(", to_expression(vec1), ")");
+		} else
```

Comment:
Minor note, but in this project, the style is to put `if`/`else` on different lines from the braces.

If you haven't already, you should install `clang-format(1)`. It comes with a Git extension command (`git-clang-format(1)`) that can run `clang-format` over only your changes.

Reply:
CLion already auto formats using clang-format for me generally. I already changed this earlier with the first block of code, but didn't in the second.